### PR TITLE
Add support for building FixedSizeListBuilder in struct_builder's mak…

### LINKED ISSUE #6594: Support DataType::FixedSizeList in make_builder within struct_builder.rs
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Add the FixedSizeList datatype to `make_builder` in `struct_builder`

**Describe the solution you'd like**
Add the appropriate code

**Describe alternatives you've considered**
Adding the appropriate code

### PR DESCRIPTION
**Pull request by kszlim**
…e_builders

# Which issue does this PR close?
Closes #6594 .

# Rationale for this change
 Makees `make_builders` more flexible/useful

# What changes are included in this PR?
Adds support for FixedSizeList builders in make_builders

# Are there any user-facing changes?
No API changes, but users will be able to successfully construct FixedSizeList builders.

### CODE REVIEW

**Review by tustvold — COMMENTED** [review]
Can we get a test please

File: arrow-array/src/builder/struct_builder.rs [hunk 1]
```diff
@@ -253,6 +253,13 @@ pub fn make_builder(datatype: &DataType, capacity: usize) -> Box<dyn ArrayBuilde
             let builder = make_builder(field.data_type(), capacity);
             Box::new(LargeListBuilder::with_capacity(builder, capacity).with_field(field.clone()))
         }
+        DataType::FixedSizeList(field, size) => {
+            let builder = make_builder(field.data_type(), capacity);
```
Comment (tustvold) [inline arrow-array/src/builder/struct_builder.rs:257]: I think this should be `capacity*size`

Comment (tustvold) [reply arrow-array/src/builder/struct_builder.rs:257]: ListBuilder doesn't have a fixed size

File: arrow-array/src/builder/struct_builder.rs [hunk 2]
```diff
@@ -253,6 +253,18 @@ pub fn make_builder(datatype: &DataType, capacity: usize) -> Box<dyn ArrayBuilde
             let builder = make_builder(field.data_type(), capacity);
             Box::new(LargeListBuilder::with_capacity(builder, capacity).with_field(field.clone()))
         }
+        DataType::FixedSizeList(field, size) => {
+            let size = *size;
+            let values_builder_capacity = {
+                let size: usize = size.try_into().unwrap();
+                capacity * size
+            };
+            let builder = make_builder(field.data_type(), values_builder_capacity);
```
Comment (alamb) [inline arrow-array/src/builder/struct_builder.rs:262]: 👍

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @kszlim

**Pull request merged**